src/components/ReportActionItem/TaskPreview.tsx

```diff
@@ -24,141 +21,135 @@ import * as LocalePhoneNumber from '@libs/LocalePhoneNumber';
+        /** The ID of the associated policy */
+        // eslint-disable-next-line
+        policyID: string;
+        /** The ID of the associated taskReport */
+        taskReportID: string;
+
+        /** Whether the task preview is hovered so we can modify its style */
+        isHovered: boolean;
+
+        /** The linked reportAction */
+        action: OnyxEntry<ReportAction>;
+
+        /** The chat report associated with taskReport */
+        chatReportID: string;
+
+        /** Popover context menu anchor, used for showing context menu */
+        contextMenuAnchor: Element;
+
+        /** Callback for updating context menu active state, used for showing context menu */
+        checkIfContextMenuActive: () => void;
+    };
+
+function TaskPreview({
+    taskReport,
+    taskReportID,
+    action,
+    contextMenuAnchor,
+    chatReportID,
+    checkIfContextMenuActive,
+    isHovered,
+    currentUserPersonalDetails,
+    rootParentReportpolicy,
+}: TaskPreviewProps) {
     const styles = useThemeStyles();
     const StyleUtils = useStyleUtils();
     const personalDetails = usePersonalDetails() || CONST.EMPTY_OBJECT;
+    const {translate} = useLocalize();
     // The reportAction might not contain details regarding the taskReport
     // Only the direct parent reportAction will contain details about the taskReport
     // Other linked reportActions will only contain the taskReportID and we will grab the details from there
-    const isTaskCompleted = !_.isEmpty(props.taskReport)
-        ? props.taskReport.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && props.taskReport.statusNum === CONST.REPORT.STATUS.APPROVED
-        : props.action.childStateNum === CONST.REPORT.STATE_NUM.SUBMITTED && props.action.childStatusNum === CONST.REPORT.STATUS.APPROVED;
-    const taskTitle = _.escape(TaskUtils.getTaskTitle(props.taskReportID, props.action.childReportName));
-    const taskAssigneeAccountID = Task.getTaskAssigneeAccountID(props.taskReport) || props.action.childManagerAccountID;
-    const assigneeLogin = lodashGet(personalDetails, [taskAssigneeAccountID, 'login'], '');
-    const assigneeDisplayName = lodashGet(personalDetails, [taskAssigneeAccountID, 'displayName'], '');
+    const isTaskCompleted = isNotEmptyObject(taskReport)
+        ? taskReport?.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && taskReport.statusNum === CONST.REPORT.STATUS.APPROVED
+        : action?.childStateNum === CONST.REPORT.STATE_NUM.SUBMITTED && action?.childStatusNum === CONST.REPORT.STATUS.APPROVED;
+    const taskTitle = Str.htmlEncode(TaskUtils.getTaskTitle(taskReportID, action?.childReportName ?? ''));
+    const taskAssigneeAccountID = Task.getTaskAssigneeAccountID(taskReport ?? {}) ?? action?.childManagerAccountID ?? '';
+    const assigneeLogin = taskAssigneeAccountID ? personalDetails[taskAssigneeAccountID]?.login ?? '' : '';
+    const assigneeDisplayName = taskAssigneeAccountID ? personalDetails[taskAssigneeAccountID]?.displayName ?? '' : '';
```

Comment:
I understand that we've said in these migrations that lodash and underscore should be removed, but I can't help but think that the end result is so much less readable and understandable than it was before. These four variables are perfect examples where it appears very confusing and the lodash implementation was much better.

I'd like to open up a discussion about this. 

cc @puneetlath @cead22 @AndrewGable @roryabraham @iwiznia

Reply:
That looks better, thanks for the suggestion!